obs/api/app/controllers/api/storage.py

```diff
@@ -266,6 +272,9 @@ def post(self, bucket_name):
         object_name = args["object_name"] if args["object_name"] else filename
 
         try:
+            regex = r"[\"\{}^%`\]\[~<>|#]|[\x80-\xFf]"
```

Comment:
regex replace matching char for **Object Name Rules** 

set [**\"\{}^%`\]\[~<>|#**]

match special char **"** (quotation)
match special char **{** (curly brace open)
match **}** (curly brace end)
match **^** (caret)
match **%** (percentage) 
match **`** (grave acent)
match special char **]** (square bracket end)
match special char **[** (square bracket opening)
match **~** (tilde)
match **<** (less than)
match **>** (greater than)
match **|** (vertical bar)
match **#** (pound)
**OR** 
set [**\x80-\xFf**]
character in the range between (index 128) and (index 255) in ASCII Code